### PR TITLE
feat(command): merge file ls and ls

### PR DESCRIPTION
**Pull request by overbool**
License: MIT
Signed-off-by: Overbool <overbool.xu@gmail.com>

Fixes: https://github.com/ipfs/go-ipfs/issues/5497 and https://github.com/ipfs/go-ipfs/issues/5502

### CODE REVIEW

File: core/commands/ls.go [hunk 1]
```diff
@@ -48,6 +48,15 @@ the following format:
...
 The JSON output contains type information.
+The path can be a prefixless ref; in this case, we assume it to be an
+/ipfs ref and not /ipns.
+
+Example:
+
+    > ipfs file ls QmW2WQi7j6c7UgJTarActp7tDNikE4B2qXtFCfLPdsgaTQ
+    cat.jpg
+    > ipfs file ls /ipfs/QmW2WQi7j6c7UgJTarActp7tDNikE4B2qXtFCfLPdsgaTQ
+    cat.jpg
```
Comment (kevina) [inline core/commands/ls.go]: I think this information is overkill.  Syntax for paths should be documented somewhere else.

File: core/commands/ls.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ import (
 type LsLink struct {
 	Name, Hash string
 	Size       uint64
-	Type       unixfspb.Data_DataType
```
Comment (kevina) [inline core/commands/ls.go]: I am not sure I like the idea of replacing an enum type with a string....

**Comment by overbool** [comment]
> We have to maintain API compatibilities so we can only add fields.

@Kubuxu Which modification do you mean?

**Comment by Kubuxu** [comment]
As an example the switch from enum to string.

**Comment by overbool** [comment]
@Kubuxu I had changed it.

**Comment by overbool** [comment]
@Kubuxu I had changed it as your suggestion and updated the test.

**Pull request closed without merging**